packages/compass-e2e-tests/smoke-test.ts

```diff
@@ -0,0 +1,338 @@
+  if (context.isWindows) {
+    return 'win32';
+  }
+
+  if (context.isOSX) {
+    return 'darwin';
+  }
+
+  if (context.isRHEL || context.isUbuntu) {
+    return 'linux';
+  }
+
+  return process.platform;
+}
+
+type PackageFilterConfig = Pick<
+  SmokeTestsContext,
+  'isWindows' | 'isOSX' | 'isRHEL' | 'isUbuntu' | 'extension'
+>;
+
+// subsets of the hadron-build info result
+
+const windowsFilenameKeys = [
+  'windows_setup_filename',
+  'windows_msi_filename',
+  'windows_zip_filename',
+  'windows_nupkg_full_filename',
+] as const;
+type WindowsBuildInfo = Record<typeof windowsFilenameKeys[number], string>;
+
+const osxFilenameKeys = ['osx_dmg_filename', 'osx_zip_filename'] as const;
+type OSXBuildInfo = Record<typeof osxFilenameKeys[number], string>;
+
+const ubuntuFilenameKeys = [
+  'linux_deb_filename',
+  'linux_tar_filename',
+] as const;
+type UbuntuBuildInfo = Record<typeof ubuntuFilenameKeys[number], string>;
+
+const rhelFilenameKeys = ['linux_rpm_filename', 'rhel_tar_filename'] as const;
+type RHELBuildInfo = Record<typeof rhelFilenameKeys[number], string>;
+
+function buildInfoIsWindows(buildInfo: any): buildInfo is WindowsBuildInfo {
+  for (const key of windowsFilenameKeys) {
+    if (!buildInfo[key]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function buildInfoIsOSX(buildInfo: any): buildInfo is OSXBuildInfo {
+  for (const key of osxFilenameKeys) {
+    if (!buildInfo[key]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function buildInfoIsUbuntu(buildInfo: any): buildInfo is UbuntuBuildInfo {
+  for (const key of ubuntuFilenameKeys) {
+    if (!buildInfo[key]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function buildInfoIsRHEL(buildInfo: any): buildInfo is RHELBuildInfo {
+  for (const key of rhelFilenameKeys) {
+    if (!buildInfo[key]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+type Package = {
+  name: string;
+  filepath: string;
+};
+
+function getFilteredPackages(
+  compassDir: string,
+  buildInfo: any,
+  config: PackageFilterConfig
+): Package[] {
+  let names: string[] = [];
+
+  if (config.isWindows) {
+    if (!buildInfoIsWindows(buildInfo)) {
+      throw new Error('missing windows package keys');
+    }
+    names = windowsFilenameKeys.map((key) => buildInfo[key]);
+  } else if (config.isOSX) {
+    if (!buildInfoIsOSX(buildInfo)) {
+      throw new Error('missing osx package keys');
+    }
+    names = osxFilenameKeys.map((key) => buildInfo[key]);
+  } else if (config.isRHEL) {
+    if (!buildInfoIsRHEL(buildInfo)) {
+      throw new Error('missing rhel package keys');
+    }
+    names = rhelFilenameKeys.map((key) => buildInfo[key]);
+  } else if (config.isUbuntu) {
+    if (!buildInfoIsUbuntu(buildInfo)) {
+      throw new Error('missing ubuntu package keys');
+    }
+    names = ubuntuFilenameKeys.map((key) => buildInfo[key]);
+  }
+
+  const extension = config.extension;
+
+  return names
+    .filter((name) => !extension || name.endsWith(extension))
```

Comment:
By optionally filtering the files we'll test we can easily run this on just one file at a time during development and we can also change the evergreen config to add one task per file if we wanted to at some point.